powerstore/resource_volume_group.go

```diff
@@ -257,4 +286,70 @@ func (r resourceVolumeGroup) updateVolGroupState(volgroupState *models.Volumegro
 		volumeIDList = append(volumeIDList, types.StringValue(string(volumeID)))
 	}
 	volgroupState.VolumeIDs, _ = types.SetValue(types.StringType, volumeIDList)
+
+	//Update VolumeNames value from Plan to State
+	var volumeNames []string
+	for _, volumeName := range volGroupPlan.VolumeNames.Elements() {
+		volumeNames = append(volumeNames, strings.Trim(volumeName.String(), "\""))
+	}
+	volumeNameList := []attr.Value{}
+	for _, volumeName := range volumeNames {
+		volumeNameList = append(volumeNameList, types.StringValue(string(volumeName)))
+	}
+	volgroupState.VolumeNames, _ = types.SetValue(types.StringType, volumeNameList)
+
+	//Update ProtectionPolicyName value from Plan to State
+	volgroupState.ProtectionPolicyName = volGroupPlan.ProtectionPolicyName
+}
+
+// ModifyPlan modify resource plan attribute value
+func (r *resourceVolumeGroup) ModifyPlan(ctx context.Context, req resource.ModifyPlanRequest, resp *resource.ModifyPlanResponse) {
+	if req.Plan.Raw.IsNull() {
+		return
+	}
+	var plan models.Volumegroup
+
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var volumeIds []string
+	if len(plan.VolumeNames.Elements()) != 0 {
+		for _, volumeName := range plan.VolumeNames.Elements() {
+			volume, err := r.client.PStoreClient.GetVolumeByName(context.Background(), strings.Trim(volumeName.String(), "\""))
+			if err != nil {
+				resp.Diagnostics.AddError(
+					"Error getting volume ",
+					"Could not get volume with name: "+strings.Trim(volumeName.String(), "\"")+", \n unexpected error: "+err.Error(),
+				)
+				return
+			}
+			volumeIds = append(volumeIds, strings.Trim(volume.ID, "\""))
+		}
+		volumeList := []attr.Value{}
+		for i := 0; i < len(volumeIds); i++ {
+			volumeList = append(volumeList, types.StringValue(string(volumeIds[i])))
+		}
+		plan.VolumeIDs, _ = types.SetValue(types.StringType, volumeList)
+	}
+
+	if plan.ProtectionPolicyName.ValueString() != "" {
+		policy, err := r.client.PStoreClient.GetProtectionPolicyByName(context.Background(), plan.ProtectionPolicyName.ValueString())
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Error getting protection policy",
+				"Could not get protection policy with name: "+plan.ProtectionPolicyName.ValueString()+", \n unexpected error: "+err.Error(),
+			)
+			return
+		}
+		plan.ProtectionPolicyID = types.StringValue(policy.ID)
+	}
+
+	diags = resp.Plan.Set(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
```

Comment:
Can you add some tests for this volume name and protection policy name ??

Reply:
@AnikaAgiwal2711 added tests for these through PR https://github.com/dell/terraform-provider-powerstore/pull/49